packages/pds/src/api/app/bsky/feed/getPostThread.ts

```diff
@@ -24,7 +24,7 @@ export default function (server: Server) {
         .executeTakeFirst()
 
       if (!queryRes) {
-        throw new InvalidRequestError(`Post not found: ${uri}`)
+        throw new InvalidRequestError(`Post not found: ${uri}`, 'NotFound')
```

Comment:
There's a case to be made that this could be a 404, but I can see it either way.  Mostly just want to surface it to ensure that option was considered 👍

Reply:
A `404` in XRPC means the XRPC method isn't implemented.

Reply:
Ah, I thought those might have been turned into 501s.

Reply:
Did a little poking around and didn't get to the bottom of it, but I did see that 404 is used by the xrpc client as `XRPCNotSupported`, I think when an unsupported status code is returned from the server.  501s are definitely in the mix too in some capacity as `MethodNotImplemented` on the server.  Anyways, I'm not too hung-up on it—this looks good!

Reply:
hah i was actually thinking this exact same thing before i got to this comment. i think partially owing to devin & i chatting earlier about trying to keep xrpc as expressive as http as possible.

Losing out on 404s is pretty unfortunate. They're like the most popular status code 😅 and 501s more accurately describe the error.

i wonder if in xrpc-server we should have a "fallback handler" that sends a 501 for any request to `/xrpc/...` that doesn't match a method

Reply:
anyway outside of the scope of this PR. but i think it's worth thinking on a bit 🤔

Reply:
ah i actually just realized that we do that exact thing already
501 - XRPC supported but not this method
404 - XRPC not supported

mer i see why that's important but still feels a shame :/

Reply:
Yeah you just have to think about how HTTP servers are going to behave generally. If the server gives a 404, it's saying it doesn't have xrpc routes.